ios/FluentUI/Core/Theme/Tokens/AliasTokens.swift

```diff
@@ -166,15 +171,36 @@ public final class AliasTokens: NSObject {
     }
     public let colors: TokenSet<ColorsTokens, DynamicColor>
 
+    // MARK: - Gradient Colors
+
+    @objc(MSFGradientColorAliasTokens)
+    public enum GradientTokens: Int, TokenSetKey {
+        case flair
+        case tint
+    }
+
+    @available(swift, obsoleted: 1.0, message: "This method exists for Objective-C backwards compatibility and should not be invoked from Swift. Please use the `gradientColors` property directly.")
+    @objc(aliasGradientForToken:)
+    public func gradient(_ token: GradientTokens) -> [UIColor] {
+        return gradientColors[token]
+    }
+    public lazy var gradientColors: TokenSet<GradientTokens, [UIColor]> = {
+        return .init(self.defaultGradientColors(_:), gradientOverrides)
+    }()
+
+    private let gradientOverrides: [GradientTokens: [UIColor]]?
+
     // MARK: Initialization
 
     init(colorOverrides: [ColorsTokens: DynamicColor]? = nil,
          shadowOverrides: [ShadowTokens: ShadowInfo]? = nil,
-         typographyOverrides: [TypographyTokens: FontInfo]? = nil) {
+         typographyOverrides: [TypographyTokens: FontInfo]? = nil,
+         gradientOverrides: [GradientTokens: [UIColor]]? = nil) {
 
         self.colors = .init(AliasTokens.defaultColors(_:), colorOverrides)
         self.shadow = .init(AliasTokens.defaultShadows(_:), shadowOverrides)
         self.typography = .init(AliasTokens.defaultTypography(_:), typographyOverrides)
+        self.gradientOverrides = gradientOverrides
```

Comment:
Is this pattern intentionally different from the others above? Where we init in the lazy getter instead of during `.init`?

Reply:
Yes, lazy initialization was the only way I could think of to ensure `gradientColors` uses the instance's `colors` tokens in order to return the correct value. Otherwise, the [other pattern](https://github.com/microsoft/fluentui-apple/pull/1777/commits/44751ddab8dd039caa6a25a265ccd5e5c40fda5a) will always return default token values even after the branding colors have been overridden by `ColorProviding`.

Reply:
I think this was mentioned in the description, but since `gradientColors` returns an array of alias tokens, it's lazy so it can pull the correct overrides.

Reply:
Is that true? The `ColorProviding` values should already be set in line 175 above, right?

```swift
self.colors = .init(AliasTokens.defaultColors(_:), colorOverrides)
```

Reply:
It was probably complaining about trying to use `self.defaultGradientColors` before calling `super.init`, and you can't get the overrides set on this theme from `AliasTokens.defaultGradientColors`.

Reply:
> Is that true? The `ColorProviding` values should already be set in line 175 above, right?
> 
> ```swift
> self.colors = .init(AliasTokens.defaultColors(_:), colorOverrides)
> ```

Right, but `self.colors` is an instance property and cannot be accessed in a static context to initialize `gradientColors`.

Reply:
we should write a comment on why this initialization is different because we can't make the graident initialization static func...